### PR TITLE
[EOS-19912] S3_MOTR_MAX_UNITS_PER_REQUEST implementation in post install phase

### PR DESCRIPTION
**Pull request by vimalraghwani20**
# Change Summary
## Problem Statement/Description
_[EOS-19912](https://jts.seagate.com/browse/EOS-19912):_
_S3_MOTR_MAX_UNITS_PER_REQUEST implementation in post install phase._

## Solution Overview

- default value of S3_MOTR_MAX_UNITS_PER_REQUEST should be 8 in to S3config.release.yaml
- Add cortx>software>s3>motr_max_units_per_request key in to S3 prov config.yaml
- Add these key in the post install template file and sample
- read cortx>software>s3>motr_max_units_per_request from provisioner config file and update it in to s3config.yaml in the post install phase

## Unit Test Cases
- Ran Post install phase on mini provisioner VM

<details>
  <summary>Output of Post_Install</summary>
[root@ssc-vm-4698 s3]# /opt/seagate/cortx/s3/bin/s3_setup post_install --config "yaml:///opt/seagate/cortx/s3/conf/s3.post_install.tmpl.1-node"
2021-06-07 03:24:49,702 - s3-deployment-logger - INFO - post_install started.
2021-06-07 03:24:49,718 - s3-deployment-logger - INFO - Processing post_install yaml:///opt/seagate/cortx/s3/conf/s3.post_install.tmpl.1-node
2021-06-07 03:24:49,718 - s3-deployment-logger - INFO - validations started
2021-06-07 03:24:49,719 - s3-deployment-logger - INFO - Validations running from /opt/seagate/cortx/s3/mini-prov/s3setup_prereqs.json
2021-06-07 03:24:51,566 - s3-deployment-logger - INFO - Validation complete
2021-06-07 03:24:51,566 - s3-deployment-logger - INFO - validating S3 config files for post_install.
2021-06-07 03:24:51,566 - s3-deployment-logger - INFO - validating config file /opt/seagate/cortx/s3/conf/s3config.yaml.
2021-06-07 03:24:51,620 - s3-deployment-logger - INFO - config file /opt/seagate/cortx/s3/conf/s3config.yaml validated successfully.
2021-06-07 03:24:51,620 - s3-deployment-logger - INFO - validating config file /opt/seagate/cortx/auth/resources/authserver.properties.
2021-06-07 03:24:51,621 - s3-deployment-logger - INFO - config file /opt/seagate/cortx/auth/resources/authserver.properties validated successfully.
2021-06-07 03:24:51,621 - s3-deployment-logger - INFO - validating config file /opt/seagate/cortx/auth/resources/keystore.properties.
2021-06-07 03:24:51,621 - s3-deployment-logger - INFO - config file /opt/seagate/cortx/auth/resources/keystore.properties validated successfully.
2021-06-07 03:24:51,621 - s3-deployment-logger - INFO - validating config file /opt/seagate/cortx/s3/s3backgrounddelete/config.yaml.
2021-06-07 03:24:51,643 - s3-deployment-logger - INFO - config file /opt/seagate/cortx/s3/s3backgrounddelete/config.yaml validated successfully.
2021-06-07 03:24:51,644 - s3-deployment-logger - INFO - validating config file /opt/seagate/cortx/s3/s3backgrounddelete/s3_cluster.yaml.
2021-06-07 03:24:51,648 - s3-deployment-logger - INFO - config file /opt/seagate/cortx/s3/s3backgrounddelete/s3_cluster.yaml validated successfully.
2021-06-07 03:24:51,648 - s3-deployment-logger - INFO - validations completed
2021-06-07 03:24:51,648 - s3-deployment-logger - INFO - update motr max units per request started
2021-06-07 03:24:51,648 - s3-deployment-logger - INFO - motr_max_units_per_request: 8
2021-06-07 03:24:51,679 - s3-deployment-logger - INFO - Key S3_MOTR_CONFIG>S3_MOTR_MAX_UNITS_PER_REQUEST updated successfully in /opt/seagate/cortx/s3/conf/s3config.yaml
2021-06-07 03:24:51,680 - s3-deployment-logger - INFO - update motr max units per request completed
2021-06-07 03:24:51,680 - s3-deployment-logger - INFO - post_install successful.
[root@ssc-vm-4698 s3]#

</details>

### CODE REVIEW

**Review by nileshgovande — APPROVED** [review]

**Review by SeagateChaDeepak — APPROVED** [review]
+1 LGTM

File: scripts/provisioning/configcmd.py [hunk 1]
```diff
@@ -57,6 +58,10 @@ def process(self, configure_only_openldap = False, configure_only_haproxy = Fals
     self.logger.info("validations completed")
 
     try:
+      self.logger.info("update motr max units per request started")
+      self.update_motr_max_units_per_request()
```
Comment (nileshgovande) [inline scripts/provisioning/configcmd.py:62]: Add basic validation

more than 1 and less than 128

Comment (vimalraghwani20) [reply scripts/provisioning/configcmd.py:62]: fixed

**Comment by vimalraghwani20** [comment]
http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Component-Test/job/S3Server-PreMerge-Test/1375/

**Comment by vimalraghwani20** [comment]
http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Component-Test/job/S3Server-PreMerge-Test/1374/

File: s3config.release.yaml.sample [hunk 2]
```diff
@@ -89,7 +89,7 @@ S3_MOTR_CONFIG:                                     # Section for S3 Motr
                                                         # 1->4K, 2->8K, 3->16K, 4->32K, 5->64K, 6->128K, 7->256K, 8->512K, 9->1024K,
                                                         # 10->2048K, 11->4096K, 12->8192K, 13->16384K, 14->32768K, default is layout id 9 (1 MB)
    S3_MOTR_UNIT_SIZE: 1048576                         # Motr unit size w.r.t layout id for an IO operation
-   S3_MOTR_MAX_UNITS_PER_REQUEST: 32                  # Maximum units per read/write request to motr
+   S3_MOTR_MAX_UNITS_PER_REQUEST: 8                  # Maximum units per read/write request to motr
```
Comment (knrajnambiar76) [inline s3config.release.yaml.sample:92]: You can add a comment here, saying that on hardware the value is set to 32, on VM/OVA its 8

Comment (vimalraghwani20) [reply s3config.release.yaml.sample]: fixed

**Comment by vimalraghwani20** [comment]
Mini Provisioner
http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Mini-Provisioning/job/cortx-s3server/1287/

**Review by sachitanands — APPROVED** [review]
looks good to me

**Comment by gowthamchinna** [comment]
test this please

**Pull request merged**